src/utils/url.web.ts

```diff
@@ -11,7 +11,11 @@ export function resolve(base: string, relative: string): string {
 
   // If the URL object is supported, we prefer that.
   if (typeof URL !== 'undefined') {
-    return (new URL(relative, base).toString());
+    try {
+      return (new URL(relative, base).toString());
+    } catch (error) {
+      // Fallback to custom resolution
+    }
```

Comment:
Is there a way to do this without eating up exceptions?

Maybe we can detect that a relative url for `base` was passed, and skip this first branch? The easiest might just be to test if the string doesn't start with http:// or https://

Reply:
I replaced with a test checking for http:// or https://